pkg/trace/obfuscate/sql.go

```diff
@@ -265,20 +265,17 @@ func (oq *ObfuscatedQuery) Cost() int64 {
 // attemptObfuscation attempts to obfuscate the SQL query loaded into the tokenizer, using the
 // given set of filters.
 func attemptObfuscation(tokenizer *SQLTokenizer) (*ObfuscatedQuery, error) {
-	filters := []tokenFilter{
```

Comment:
Why did you change this logic? Did it bring any benefit?

Reply:
Yes, when profiling, this showed up as a significant source of allocations - allocating an array, several objects, and then interfaces for them for each query.

I didn't record exactly how much this specific change improved performance, but I thought it was worthwhile.

Reply:
The overhead ends up being pretty significant.

I ran this benchmark against attemptObfuscation using the filter slice method (`oldattempt.txt`) and the one in this PR (`newattempt.txt`). using `go test -v -run NONE -bench BenchmarkObfuscateSQLString -benchmem -benchtime 30s`
Everything else was the same in both (including the `out = ` change)
```
% benchcmp oldattempt.txt newattempt.txt
benchmark                                      old ns/op     new ns/op     delta
BenchmarkObfuscateSQLString/Escaping/512-8     9594          7894          -17.72%
BenchmarkObfuscateSQLString/Grouping/199-8     4945          3791          -23.34%
BenchmarkObfuscateSQLString/query1/3694-8      83517         69018         -17.36%
BenchmarkObfuscateSQLString/query3/980-8       23150         17440         -24.67%
BenchmarkObfuscateSQLString/random-8           1596          1157          -27.51%

benchmark                                      old allocs     new allocs     delta
BenchmarkObfuscateSQLString/Escaping/512-8     25             12             -52.00%
BenchmarkObfuscateSQLString/Grouping/199-8     24             10             -58.33%
BenchmarkObfuscateSQLString/query1/3694-8      152            83             -45.39%
BenchmarkObfuscateSQLString/query3/980-8       13             6              -53.85%
BenchmarkObfuscateSQLString/random-8           11             8              -27.27%

benchmark                                      old bytes     new bytes     delta
BenchmarkObfuscateSQLString/Escaping/512-8     2011          1936          -3.73%
BenchmarkObfuscateSQLString/Grouping/199-8     892           816           -8.52%
BenchmarkObfuscateSQLString/query1/3694-8      16387         16256         -0.80%
BenchmarkObfuscateSQLString/query3/980-8       3077          3008          -2.24%
BenchmarkObfuscateSQLString/random-8           320           248           -22.50%
```